test/utils_test.js

```diff
@@ -306,6 +307,210 @@ describe("utils", () => {
+
+      it("should merge schema from additionalItems defaults into property default", () => {
+        const schema = {
+          type: "object",
+          properties: {
+            level1: {
+              type: "array",
+              default: [
+                {
+                  item: "property-default-1",
+                },
+                {},
+              ],
+              additionalItems: {
+                type: "object",
+                properties: {
+                  item: {
+                    type: "string",
+                    default: "additional-default",
+                  },
+                },
+              },
+              items: [
+                {
+                  type: "object",
+                  properties: {
+                    item: {
+                      type: "string",
+                    },
+                  },
+                },
+              ],
+            },
+          },
+        };
+
+        expect(getDefaultFormState(schema, {})).eql({
+          level1: [
+            { item: "property-default-1" },
+            { item: "additional-default" },
+          ],
+        });
+      });
+
+      it("should overwrite defaults over multiple levels with arrays", () => {
+        const schema = {
+          type: "object",
+          default: {
+            level1: [
+              {
+                item: "root-default-1",
+              },
+              {
+                item: "root-default-2",
+              },
+              {
+                item: "root-default-3",
+              },
+              {
+                item: "root-default-4",
+              },
+            ],
+          },
+          properties: {
+            level1: {
+              type: "array",
+              default: [
+                {
+                  item: "property-default-1",
+                },
+                {},
+                {},
+              ],
+              additionalItems: {
+                type: "object",
+                properties: {
+                  item: {
+                    type: "string",
+                    default: "additional-default",
+                  },
+                },
+              },
+              items: [
+                {
+                  type: "object",
+                  properties: {
+                    item: {
+                      type: "string",
+                    },
+                  },
+                },
+                {
+                  type: "object",
+                  properties: {
+                    item: {
+                      type: "string",
+                      default: "child-default-2",
+                    },
+                  },
+                },
+              ],
+            },
+          },
+        };
+
+        expect(getDefaultFormState(schema, {})).eql({
+          level1: [
+            { item: "property-default-1" },
+            { item: "child-default-2" },
+            { item: "additional-default" },
+          ],
```

Comment:
Can you add a test in which you make sure that one of the root-defaults are injected? (Similar to the grandparent test, but a bit simpler since it's only one level)

Reply:
I am not sure I understand what we should test there additionally which is not covered yet by the grandparent test?

Reply:
I had meant to have a "parent" test instead of a "grandparent" test. You are right, though, the grandparent test already seems to cover that logic.